From 6d7d00d49ae1ab54d66d24621c01b59b8bd68f43 Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Mon, 15 Jan 2024 17:36:25 +0100
Subject: [PATCH 0576/1695] plat-stm32mp1: call do_reset for PSCI system reset

Call the new function do_reset() in PSCI system reset function.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 152144c8fc237e5d7fbc60c254162ce8cf94c8f3 plat-stm32mp1: call do_reset for PSCI system reset]
Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Change-Id: I2da9e50a093ed6a0c5c0afa547c3c976c032d921
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/351402
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Tested-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 core/arch/arm/plat-stm32mp1/pm/psci.c | 7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/core/arch/arm/plat-stm32mp1/pm/psci.c b/core/arch/arm/plat-stm32mp1/pm/psci.c
index 38b4f87fc..b02b0aa65 100644
--- a/core/arch/arm/plat-stm32mp1/pm/psci.c
+++ b/core/arch/arm/plat-stm32mp1/pm/psci.c
@@ -408,12 +408,7 @@ void __noreturn psci_system_off(void)
 /* Override default psci_system_reset() with platform specific sequence */
 void __noreturn psci_system_reset(void)
 {
-	IMSG("Forced system reset");
-	console_flush();
-	mdelay(10);
-	stm32_reset_system();
-	udelay(100);
-	panic();
+	do_reset("PSCI");
 }
 
 /* Override default psci_cpu_on() with platform supported features */
-- 
2.47.2

